datajunction-server/datajunction_server/models/history.py

```diff
@@ -18,14 +18,26 @@ class ActivityType(str, Enum):
     An activity type
     """
 
+    AVAILABILITY = "availability"
     CREATE = "create"
     DEACTIVATE = "deactivate"
     ACTIVATE = "activate"
     UPDATE = "modify"
+
+    # Link columns to dimensions
     LINK = "link"
+    UNLINK = "unlink"
```

Comment:
I know `LINK` was already under activity type but looking at it now I think it should be an entity type and we should have a `DELETE` activity type. Then the history could be captured as `activity_type=ActivityType.[DELETE|CREATE], entity_type=EntityType.LINK`?

Reply:
Yeah, I thought about structuring it the way you described, but then I realized that when we ask for the history of an entity like `node`, we won't get a lot of the other activities that are closely related to nodes like linking a dimension or adding a materialization in a single query. I guess this is thinking of the EntityType as the level at which we want to retrieve user activity, rather than a direct mapping to all of the entities we actually store in the backend. Otherwise we'd take a performance hit if we want to actually display all activity related to a node, as we'd need to hit the backend 3+ times to get node-level, link-level, and materialization-level activity. But maybe that doesn't matter too much?

Reply:
That's a good point, I don't think the performance is as much of an issue since the selects should remain very simple, but we do want to make it as convenient as possible to query. Maybe we can add a "context_node" attribute that always includes the relevant node name? That should easily support these query patterns:
- Get all activity for a node
- Get all of a type of activity (i.e. show me global create activity)
- Get all activity for an entity type (i.e. show me global activity for materializations)

What do you think?

Reply:
This is a tough one for me too but I am leaning towards what @samredai is saying and I think a cleaner model will be better in the long term than the ability to query "everything" in one swoop. I think we'll be ok with the ability to query widely "by even type" and just add some consolidation at the places where we need, say if we ever wanted to do "get_everything_about_a_node", but one node, not all of them.

Reply:
Yeah, good point, we can probably rework the history endpoint if needed to provide node-level activity, which would include all the links, materializations etc) and not store the `context_node` at all.